### ISSUE TITLE
Error when disabling javaTimeTypes on table valued function returning TIMESTAMPTZ type

### ISSUE THREAD

**Issue by kochetkovSergeY** [jOOQ/jOOQ#11595]
### Expected behavior
 java.time.* column types won't be generated when use setting withJavaTimeTypes(false)

```java
            .withGenerate(new Generate()
                .withIndexes(false)
                .withJavaTimeTypes(false))
```

### Actual behavior
GenerationTool still generating java.time.OffsetDateTime with .withJavaTimeTypes(false)

### Steps to reproduce the problem

```java
   Path pathToGenerate = Paths.get("C:\\path_to_generate");
    List<String> schemaNames = Arrays.asList("schema1");
    List<SchemaMappingType> schemaList = schemaNames.stream()
        .map(it -> new SchemaMappingType().withInputSchema(it)).collect(Collectors.toList());

    Configuration configuration = new Configuration()
        .withLogging(Logging.DEBUG)
        .withJdbc(new Jdbc()
            .withDriver("org.postgresql.Driver")
            .withUrl("jdbc:postgresql://host:5432/db")
            .withUser("username")
            .withPassword("password"))
        .withGenerator(new Generator()
            .withName(JavaGenerator.class.getName())
            .withDatabase(new Database()
                .withSchemata(schemaList)
                .withForcedTypes(
                    new ForcedType()
                        .withName(SQLDataType.TIMESTAMP.getTypeName())
                        .withIncludeExpression(".*")
                        .withIncludeTypes("^" + SQLDataType.TIMESTAMP.getTypeName() + ".*")
                        .withNullability(Nullability.ALL)
                        .withObjectType(ForcedTypeObjectType.COLUMN)
                )
            )
            .withGenerate(new Generate()
                .withIndexes(false)
                .withJavaTimeTypes(false))
            .withStrategy(new Strategy()
                .withName(AlwaysSchemaNameGeneratorStrategy.class.getName()))
            .withTarget(new Target()
                .withPackageName("com.test.beans")
                .withDirectory(pathToGenerate.toString())));

    GenerationTool generationTool = new GenerationTool();
    generationTool.run(configuration);
```

### Versions
- jOOQ: 3.14.8
- Java: 1.8
- Database (include vendor): PostgreSql 9.6
- OS: Windows 10
- JDBC Driver (include name if inofficial driver):postgresql-42.0.0.jar

**Comment by lukaseder** [comment]
Thank you very much for your report. Yes, that's what we're doing, see the rationale here:
https://www.jooq.org/doc/dev/manual/code-generation/codegen-advanced/codegen-config-generate/codegen-generate-java-time-types/

> Semantically, the above types are exactly equivalent, although the new types do away with the many flaws of the JDBC types. If there is no JDBC type for an equivalent JSR 310 type, then the JSR 310 type is generated by default. This includes
> 
> - java.time.OffsetTime (for SQL TIME WITH TIME ZONE)
> - java.time.OffsetDateTime (for SQL TIMESTAMP WITH TIME ZONE)
>
> To get more fine-grained control of the above, you may wish to consider applying data type rewriting.

The `ForcedType` regular expressions are case sensitive, so try using this instead:

```java
.withIncludeTypes("^(?i:timestamp.*)$")
```

I hope this helps

**Comment by lukaseder** [comment]
Ugh. That would be a bug, then. Can you please post:

- Your DDL statement of the class that produces this issue
- The contents of `AlwaysSchemaNameGeneratorStrategy`

**Comment by kochetkovSergeY** [comment]
The contents of AlwaysSchemaNameGeneratorStrategy:
```
import org.jooq.meta.CatalogDefinition;
import org.jooq.codegen.DefaultGeneratorStrategy;
import org.jooq.meta.Definition;

/**
 * jOOQ generator strategy that always appends schema name to the target package. jOOQ does do it
 * only when multiple schemas are present
 *
 * TODO remove when jOOQ 4, see jooq/jooq#3960 on github
 */
public class AlwaysSchemaNameGeneratorStrategy extends DefaultGeneratorStrategy {

  @Override
  public String getJavaPackageName(Definition definition, Mode mode) {
    String pkg = super.getJavaPackageName(definition, mode);

    if(!(definition instanceof CatalogDefinition)) {
      String subpackage = pkg.replace(getTargetPackage(), "");

      if(definition.getDatabase().getSchemata().size() == 1) {
        subpackage = "." + getJavaIdentifier(definition.getSchema()).toLowerCase() + subpackage;
      }
      pkg = getTargetPackage() + subpackage;
    }

    return pkg;
  }
}
```

**Comment by lukaseder** [comment]
Thanks, and the DDL (`CREATE TABLE`) of the table causing this issue? It will be quite helpful for me to try to reproduce this problem.

**Comment by kochetkovSergeY** [comment]
I'll post ddl tomorrow

**Comment by kochetkovSergeY** [comment]
```
-- DROP FUNCTION test_schema.test(bigint, bigint, character varying, timestamp with time zone, timestamp with time zone, bigint, timestamp with time zone, timestamp with time zone, character varying, character varying, character varying, character varying, boolean, character varying, character varying, character varying, date, date, text, boolean, character varying, character varying, integer, integer);

CREATE OR REPLACE FUNCTION test_schema.test(
	param1 bigint,
	param2 bigint,
	param3 character varying,
	param4 timestamp with time zone,
	param5 timestamp with time zone,
	param6 bigint,
	param7 timestamp with time zone,
	param8 timestamp with time zone,
	param9 character varying,
	param10 character varying,
	param11 character varying,
	param12 character varying,
	param13 boolean,
	param14 character varying,
	param15 character varying,
	param16 character varying,
	param17 date,
	param18 date,
	param19 text,
	param20 boolean,
	param21 character varying,
	param22 character varying,
	param23 integer,
	param24 integer)
    RETURNS TABLE(test1 bigint, test2 character varying, test3 timestamp with time zone, test4 character varying, test5 character varying, test6 character varying, test7 character varying, test8 timestamp with time zone, test9 text, test10 character varying, test11 character varying, test12 text, test13 text, test14 text, test15 date, test16 character varying, test17 character varying, test18 boolean, test19 timestamp without time zone, test20 character varying) 
    LANGUAGE 'plpgsql'
    COST 100
    VOLATILE PARALLEL UNSAFE
    ROWS 1000

AS $BODY$
DECLARE
	Request_str text;
BEGIN

Request_str := '
	select
		null::bigint as "test1",
		null::character varying as "test2",
		null::timestamp with time zone as "test3",
		null::character varying as "test4",
		null::character varying as "test5",
		null::character varying as "test6",
		null::character varying as "test7",
		null::timestamp with time zone as "test8",
		null::text as "test9",
		null::character varying as "test10",
		null::character varying as "test11",
		null::text as "test12",
		null::text as "test13",
		null::text as "test14",
		null::date as "test15",
		null::character varying as "test16",
		null::character varying as "test17",
		null::boolean as "test18",
		null::timestamp without time zone as "test19",
		null::character varying as "test20"
';

return query execute Request_str;

END;
$BODY$;

ALTER FUNCTION test_schema.test(bigint, bigint, character varying, timestamp with time zone, timestamp with time zone, bigint, timestamp with time zone, timestamp with time zone, character varying, character varying, character varying, character varying, boolean, character varying, character varying, character varying, date, date, text, boolean, character varying, character varying, integer, integer)
    OWNER TO zags;

```

**Comment by lukaseder** [comment]
Thanks for the update. Oh a table valued function, that seems quite relevant... I'll try to reproduce this again.

**Comment by lukaseder** [comment]
Does it still happen when you remove: `.withObjectType(ForcedTypeObjectType.COLUMN)`?

**Comment by kochetkovSergeY** [comment]
No, it's not. Thank you for help

**Comment by lukaseder** [comment]
I think that's the bug here, which should be fixed.

**Comment by lukaseder** [comment]
I've had another look at this issue. I cannot reproduce it. I've tried:

- Deactivating `javaTimeTypes`
- Adding a `forcedType` similar to yours
- Creating a table valued function with `TIMESTAMP WITH TIME ZONE` types

If you think it still persists, a reproducer based on our templates would certainly be helpful: https://github.com/jOOQ/jOOQ-mcve

Closing this issue until a reproducer is available. The issue may as well have been fixed elsewhere in the meantime, though.
